File: merchantdemo/src/main/java/com/example/merchantdemo/service/MerchantService.java
package com.example.merchantdemo.service;

import com.example.merchantdemo.dto.MerchantDto;
import com.example.merchantdemo.entity.Merchant;
import com.example.merchantdemo.entity.Store;
import com.example.merchantdemo.entity.User;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface MerchantService {
    ResponseEntity<Merchant> getMerchantById(int merchantId);

    ResponseEntity<List<Merchant>> getMerchantsByUserId(int userId);

    ResponseEntity<String> addMerchant(MerchantDto theMerchant);

    ResponseEntity<Merchant> updateMerchant(MerchantDto theMerchant);


    ResponseEntity<String> changeMerchantStore(int theMerchantId, int theStoreNumber);


    ResponseEntity<List<Merchant>> getAllMerchants();
}

File: merchantFrontEnd/src/Service/MerchantService.js
import axios from "axios";

const MERCHANT_API_BASE_URL = "http://localhost:8080/merchant";
class MerchantService {
  getAllMerchants() {
    return axios.get(MERCHANT_API_BASE_URL + "/getAll");
  }
  getMerchantsByUserId(userId) {
    return axios.get(MERCHANT_API_BASE_URL + "/all/" + userId);
  }

  getMerchantById(merchantId) {
    return axios.get(MERCHANT_API_BASE_URL + "/" + merchantId);
  }

  addMerchant(theMerchant) {
    return axios.post(MERCHANT_API_BASE_URL, theMerchant);
  }

  updateMerchant(theMerchant) {
    return axios.put(MERCHANT_API_BASE_URL, theMerchant);
  }

  changeMerchantStore(theMerchantId, theStoreNumber) {
    return axios.put(
      MERCHANT_API_BASE_URL +
        "/changeStore?theMerchantId=" +
        theMerchantId +
        "&theStoreNumber=" +
        theStoreNumber
    );
  }
}

export default new MerchantService();
